### ISSUE TITLE
Some 2 operand instructions store the result in the wrong parameter

### ISSUE THREAD

**Issue by blubberdiblub** [Spaceman1701/LudumDare38#1]
`Shl`, `Shr`, `Or`, `And` and `Xor` should all store their result in the first parameter, not in the second.

The manual says the result goes into the first parameter.
The syntax of the instruction set this is based on (x86) suggests they should go into the first parameter.
Consistency with the other instructions (`Add`, `Sub`) suggests they should go into the first parameter.
For `Shl` and `Shr` it doesn't even make a lot of sense to store the result where the number of bits to shift was before.

**Comment by Spaceman1701** [comment]
Yeah, you're completely right. Those are bugs. I might fix that on the main branch, as that seems rather game-breaking to me. Otherwise I'll fix it on the post-jam branch.
